### PR TITLE
Remove dependency of `json-serialize-refs`

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Description

Take the main logic from https://github.com/devel0/json-serialize-refs/blob/master/src/lib/stringify-refs.ts to make it under our maintenance to avoid possible supply chain attacks considering it has not been updated for a few years.

Modifications:
- Change the id to start from `0` so that the json object itself should have an id of `0` and everything we care starting from `1`.
- Remove the currently unnecessary option to reduce complexity.

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/json-serialize-refs.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+function createRefs(obj: any): any {
```
Comment (lirenhe) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/json-serialize-refs.ts:1]: Can we add our license at the begin of this file?

File: src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts [hunk 2]
```diff
@@ -27,6 +26,7 @@ import {
     resolveOutputFolder
 } from "./options.js";
 import { Configuration } from "./type/configuration.js";
+import { stringifyRefs } from "./lib/json-serialize-refs.js";
```
Comment (lirenhe) [inline src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts:29]: Can we find some other name instead of using  'json-serialize-refs' and 'stringifyRefs'

Comment (ArcturusZhang) [reply src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts:29]: do you have suggestions?

Comment (lirenhe) [reply src/TypeSpec.Extension/Emitter.Csharp/src/emitter.ts:29]: If this is not a new implementation, I suggest follow the original license in our work and there is no need for naming update.

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/json-serialize-refs.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (jsquire) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/json-serialize-refs.ts:1]: @ArcturusZhang :  I'm not 100% sure this is allowable.   Please reach out to our PM team and collaborate with them to engage the MS Open Source office for guidance.

**Pull request closed without merging**